### DISCUSSION TITLE
How to iterate the array using the int type?

### DISCUSSION THREAD

**Discussion by Kolhun** [nlohmann/json#4156] [Q&A]
Hello everyone, there is a need to iterate over the values stored in json using int while having an output, for example, as data[row] or something similar. Here is an example of data[row] and data[row].value causes an error. Is there a working alternative?
![image](https://github.com/nlohmann/json/assets/36815965/d70741b5-8e29-4dc6-94b6-57418148b380)

```c
json data = dataResult["data"];
ImGuiListClipper clipper;
clipper.Begin(data.size());

while (clipper.Step())
{
    for (int row = clipper.DisplayStart; row < clipper.DisplayEnd; row++)//for(auto& row = data.begin(); row != data.end(); ++row)//for (int row = clipper.DisplayStart; row < clipper.DisplayEnd; row++)//for (auto row = data.begin(); row != data.end(); ++row)
    {
        ImGui::TableNextRow();
        for (int column = 0; column < 4; column++)
        {
            ImGui::TableSetColumnIndex(column);
            json dataResultList;// = row.value();

            for (auto& element : data)
            {
                //std::cout << element << '\n';
                dataResultList = element;
                break;
            }
```

![image](https://github.com/nlohmann/json/assets/36815965/ae8ae378-9a5d-4a81-8e16-d21e19da502d)

**Comment by nlohmann** [comment]
For a json array, operator[] accepts an integer index directly and returns the element itself — use data[row], not data[row].value() (which is only meaningful on the object returned by an iterator's .value(), not on a direct index into the array). So your loop body should read: json element = data[row]; and then access element["someKey"] etc. data[row].value() doesn't compile/behave as intended because array elements returned by operator[](size_type) aren't iterator-proxy objects.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
